test/distributed/_tensor/test_dtensor.py

```diff
@@ -742,6 +742,18 @@ def test_redistribute_sub_mesh(self):
             mesh.mesh, torch.ones(4, 3), torch.tensor([]), sharded_again.to_local()
         )
 
+    @with_comms
+    def test_implicit_replication(self):
+        mesh = init_device_mesh(self.device_type, (self.world_size,))
+        local_tensor1 = torch.ones(4, 3)
+        sharded_dtensor = DTensor.from_local(local_tensor1, mesh, [Shard(0)])
+
+        from torch.distributed._tensor.experimental import implicit_replication
+
+        with implicit_replication():
+            out_dt = sharded_dtensor + torch.randn(3, device=self.device_type)
+            self.assertEqual(out_dt.placements, [Shard(0)])
```

Comment:
Should we include any numeric test to check the result of the add?

What is the expected semantics of adding a replicated `(3,)` local tensor to a `(4, 3)` local tensor? It should broadcast over dim-1?

Reply:
sure!

> What is the expected semantics of adding a replicated (3,) local tensor to a (4, 3) local tensor? It should broadcast over dim-1?

Yep it would be implicitly broadcasted according on the "4" dim according to the tensor broadcasting rule https://pytorch.org/docs/stable/notes/broadcasting.html#broadcasting-semantics